dmscripts/notify_suppliers_of_new_questions_answers.py

```diff
@@ -0,0 +1,159 @@
+import dmapiclient
+
+from datetime import datetime, date, timedelta
+
+from dmscripts.helpers import env_helpers, logging_helpers
+from dmscripts.helpers.html_helpers import render_html
+from dmscripts.helpers.logging_helpers import logging
+from dmutils.email.dm_mandrill import send_email
+from dmutils.formats import DATETIME_FORMAT
+
+
+EMAIL_SUBJECT = "New question and answer responses on the Digital Marketplace"
+EMAIL_FROM_ADDRESS = "do-not-reply@digitalmarketplace.service.gov.uk"
+EMAIL_FROM_NAME = "Digital Marketplace Team"
+
+
+logger = logging_helpers.configure_logger({'dmapiclient': logging.INFO})
+
+
+def get_live_briefs_with_new_questions_and_answers_between_two_dates(data_api_client, start_date, end_date):
+    briefs = data_api_client.find_briefs_iter(status='live', human=True, with_clarification_questions=True)
+
+    # return a list of briefs that contain clarification questions published between the start data and the end date
+    return [brief for brief in briefs if brief.get('clarificationQuestions') and any(
+        datetime.strptime(question['publishedAt'], DATETIME_FORMAT) >= start_date
+        and datetime.strptime(question['publishedAt'], DATETIME_FORMAT) <= end_date
+        for question in brief['clarificationQuestions']
+    )]
+
+
+def get_ids_of_suppliers_who_started_applying(data_api_client, brief):
+    responses = data_api_client.find_brief_responses(brief_id=brief["id"])
+    return [response["supplierId"] for response in responses["briefResponses"]]
+
+
+def get_ids_of_suppliers_who_asked_a_clarification_question(data_api_client, brief):
+    audit_events = data_api_client.find_audit_events(
+        audit_type=dmapiclient.audit.AuditTypes.send_clarification_question,
+        object_type='briefs',
+        object_id=brief['id']
+    )
+    return [audit_event['data']['supplierId'] for audit_event in audit_events['auditEvents']]
+
+
+def get_ids_of_interested_suppliers_for_briefs(data_api_client, briefs):
+    interested_suppliers = {}
+    for brief in briefs:
+        suppliers_who_applied = get_ids_of_suppliers_who_started_applying(data_api_client, brief)
+        suppliers_who_asked_a_question = get_ids_of_suppliers_who_asked_a_clarification_question(data_api_client, brief)
+        interested_suppliers[brief['id']] = list(set(suppliers_who_applied + suppliers_who_asked_a_question))
+
+    return invert_a_dictionary_so_supplier_id_is_key_and_brief_id_is_value(interested_suppliers)
```

Comment:
Could this just be `return dict((v,k) for k,v in interested_suppliers.iteritems())`?  Or is there more going on?  Something to do with duplicate values?